### PR TITLE
Olwa pr

### PR DESCRIPTION
**Pull request by oliver-walker-hms**
Mostly documentation changes. There were two small issues with the previous pr that are fixed in this. I also included the example configuration in the release zip.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -46,12 +46,8 @@ User configuration of the application parameters is done through the ConnectorCo
 
 #### Web Configuration
 
-The ConnectorConfig.json file can be edited through the configuration webpage.  The configuration webpage can be accessed after the [installation process](#Installation) has been completed.  The url for the configuration interface **Your_Flexy's_IP_Address**/usr/config.html
-
-![osisoft web configuration interface](docs/images/configurationPage.png)
-
 #### FTP Configuration
-If you wish to manually edit your ConnectorConfig.json file or you have a saved ConnectorConfig.json file from another Flexy you can load your ConnectorConfig.json file on the flexy in the /usr/ directory via FTP.
+If To change configuration values, manually edit your ConnectorConfig.json file on the Flexy in the /usr/ directory via FTP.
```
Comment (alexjhawk) [inline README.md:66]: The phrase "If To change configuration values, m....." doesn't make sense

File: README.md [hunk 2]
```diff
@@ -100,7 +100,7 @@ static String eWONCertificatePath = "/usr/Certificates";
 ## Configuration File
 
 ### Example ConnectorConfig.json
-```
+``` JSON
```
[the remark this answers is not in the document]
Comment (alexjhawk) [reply README.md:119]: See: https://www.markdownguide.org/extended-syntax/#syntax-highlighting

File: README.md [hunk 3]
```diff
@@ -186,6 +187,22 @@ The following are rules to make a valid tag name scheme:
...
 
+#### LoggingLevel
+The OSIsoft connector uses the [HMS Solution Center logging library](https://github.com/hms-networks/sc-flexy-logger-lib) for application logging to the Ewon Flexy's realtime logs.
+
+To change the logging level in the configuration file, specify the logging level under the AppConfig object.
+LoggingLevel is an optional field in the configuration file and is not required by the application to properly run.
+
+Below is a partial example of the configuration file set to log general application state information.
+See [Log Level](https://github.com/hms-networks/sc-flexy-logger-lib#log-level) for more information.
+   ``` JSON
```
Comment (alexjhawk) [inline README.md:215]: same suggestion about using ```json

File: .github/workflows/release-build-outputs.yml [hunk 4]
```diff
@@ -15,7 +15,7 @@ on:
 env:
   GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
   CHANGELOG_FILE_NAME: ./docs/CHANGELOG.md
-  RELEASE_INCLUDED: $CHANGELOG_FILE_NAME README.md LICENSE src .classpath .project pom.xml build.xml ./scripts/jvmrun
+  RELEASE_INCLUDED: $CHANGELOG_FILE_NAME README.md LICENSE src .classpath .project pom.xml build.xml ./scripts/jvmrun ./config/ConnectorConfig.json
```
Comment (alexjhawk) [inline .github/workflows/release-build-outputs.yml:18]: It may be worth looking in to updating this connector to use a `starting-files` folder eventually. For example, the starter project follows that structure: https://github.com/hms-networks/sc-java-maven-starter-project, as well as https://github.com/hms-networks/flexy-cumulocity-connector

**Review by alexjhawk — CHANGES_REQUESTED** [review]

File: .github/workflows/release-build-outputs.yml (hunk 4 above)
Comment (oliver-walker-hms) [reply .github/workflows/release-build-outputs.yml:18]: a good suggestion for next time. I'll make a new issue for this

File: README.md [hunk 5]
```diff
@@ -248,12 +296,12 @@ The Client ID and the Client Secret will be needed for the basic script you will
...
    1. Replace the yyy below with your tenant ID.
    1. Copy these into the ServerConfig portion of the json file.
-      ```
+      ``` json
       "Namespace":"xxx",
       "TenantId":"yyy"
       ```
    1. Make sure that the JSON formatting is valid. Below is an example of a properly formatted configuration file.
-   ```
+   ``` json
```
Comment (alexjhawk) [inline README.md:304]:
```suggestion
   ```json
```

Comment (alexjhawk) [reply README.md:304]: There are a few instances of this still remaining.

**Review by alexjhawk — CHANGES_REQUESTED** [review]

File: README.md [hunk 6]
```diff
@@ -130,6 +131,7 @@ This value can be attained by pointing a browser to the OSIsoft PIWEBAPI URL (ht
 (If the PIWEBAPI certificates are self-signed a security warning may occur when trying to access this page. This warning should be ignored)
 
 #### Proxy URL
+
```
Comment (alexjhawk) [inline README.md:151]: Adding back the whitespace you removed in previous commit be77d9a

**Review by alexjhawk — CHANGES_REQUESTED** [review]

File: README.md [hunk 7]
```diff
@@ -105,72 +116,109 @@ static String eWONCertificatePath = "/usr/Certificates";
...
 
-CertificatePath - Path to the directory containing your server's certificate.  For more information see the Certificates section
+#### CertificatePath
+Path to the directory containing your server's certificate.  For more information see the Certificates section
 
 ### AppConfig
 
-CycleTimeMs - Cycle time of the application.  All tags will be posted at this specified interval
+#### CommunicationType
+This refers to if you are using OMF (OSIsoft message format) or PIWEBAPI. OMF is the recommended setting.
```
Comment (TomKimsey) [inline README.md:178]:
```suggestion
This refers to if you are using OMF (OSIsoft Message Format) or PIWEBAPI. OMF is the recommended setting. 
```

File: README.md [hunk 8]
```diff
@@ -105,72 +116,109 @@ static String eWONCertificatePath = "/usr/Certificates";
...
+LoggingLevel is an optional field in the configuration file and is not required by the application to properly run.
+
+Below is a partial example of the configuration file set to log general application state information.
+See [Log Level](https://github.com/hms-networks/sc-flexy-logger-lib#log-level) for more information.
+   ```json
+   {
+   "AppConfig": {
+       "LoggingLevel": 4
+       }
+   }
```
Comment (TomKimsey) [inline README.md:219]:
```suggestion
   {
       "AppConfig": {
           "LoggingLevel": 4
       }
   }
```

File: src/main/java/com/hms/flexyosisoftconnector/configuration/OSIsoftTagNamingScheme.java [hunk 9]
```diff
@@ -130,12 +130,17 @@ public static String extractTagNameFromComponents(String tagName, String tagType
...
 
-    if (shortHandOSIsoftNameOptionSelected.length() <= lengthOfSingleOption) {
+    if (shortHandOSIsoftNameOptionSelected.length() >= lengthOfSingleOption) {
       // loop through every option present
       for (int index = 0;
           index < shortHandOSIsoftNameOptionSelected.length();
           index += lengthOfSingleOption) {
-        String shortHandOption = shortHandOSIsoftNameOptionSelected.substring(index, index + 1);
+
+        // add a optionNameDeliminator between name components in the OSIsoft tag name
```
Comment (TomKimsey) [inline src/main/java/com/hms/flexyosisoftconnector/configuration/OSIsoftTagNamingScheme.java:139]:
```suggestion
        // add an optionNameDeliminator between name components in the OSIsoft tag name

```

File: README.md [hunk 10]
```diff
@@ -105,72 +116,109 @@ static String eWONCertificatePath = "/usr/Certificates";
...
 
-Adding the field "ProxyURL" under the server config section will change request endpoints to https://domain-name/proxy-path. This field is optional. 
-Below is an example config file with the optional proxy URL in use.
-   ```
+This optional field is only needed when using a non default endpoint of https://HOSTNAME/piwebapi. Adding the field "ProxyURL" under the server config section will change request endpoints to https://HOSTNAME/proxy-path. HOSTNAME will be retrieved from the URL field in the configuration file. 
+
+Below is a partial config file with the optional proxy URL in use. Its endpoint is set to https://192.168.1.1/long/proxy/example/omf
+   ```json
    {
    "ServerConfig":{
```
Comment (TomKimsey) [inline README.md:156]: Not part of this PR but this should be indented 1 level.

File: README.md [hunk 11]
```diff
@@ -105,72 +116,109 @@ static String eWONCertificatePath = "/usr/Certificates";
...
 
-URL – IP address or domain name of the system running the Web API (in this example assume the IP address is 192.168.0.124)
+#### URL
+IP address or domain name of the system running the Web API (in this example assume the IP address is 192.168.0.124)
 
-WebID – WebID of your OSIsoft Web API.  This value can be attained by pointing a browser to the OSIsoft PIWEBAPI URL (https://192.168.0.124/piwebapi/dataservers).  The WebID is one of several parameters returned.  A username and password will be required to access the page.  If a username is not required it may be cached.  Try clearing the cache to verify the username and password.  The same username and password will be used to create the credentials in the next step.  (A security warning may occur when trying to access this page, this warning should be ignored)
+#### WebID
+This field is only needed when the CommunicationType is set to "piwebapi". WebID is the web ID of your OSIsoft PI Web API.
```
Comment (TomKimsey) [inline README.md:146]:
```suggestion
This field is only needed when the CommunicationType is set to "piwebapi". WebID is the web ID of the OSIsoft PI Web API instance.
```

File: README.md [hunk 12]
```diff
@@ -40,21 +54,18 @@ More information can be found here: [Ewon Historical Logging](https://developer.
...
+### Basic authentication
 
-#### Web Configuration
+This connector uses Basic authentication to communicate with OSIsoft and requires Basic authentication is enabled.
 
-The ConnectorConfig.json file can be edited through the configuration webpage.  The configuration webpage can be accessed after the [installation process](#Installation) has been completed.  The url for the configuration interface **Your_Flexy's_IP_Address**/usr/config.html
+### Configuration
 
-![osisoft web configuration interface](docs/images/configurationPage.png)
+User configuration of the application parameters is done through the ConnectorConfig.json file.  This file must be modified to match your configuration.
```
Comment (TomKimsey) [inline README.md:63]:
```suggestion
User configuration of the application parameters is done through the ConnectorConfig.json file.  This file must be modified to match the configuration.
```

```suggestion
User configuration of the application parameters is done through the ConnectorConfig.json file.  This file must be modified to match the OSIsoft configuration.
```

File: README.md [hunk 13]
```diff
@@ -40,21 +54,18 @@ More information can be found here: [Ewon Historical Logging](https://developer.
 
 ## Getting Started
 
-### Configuration
-
-User configuration of the application parameters is done through the ConnectorConfig.json file.  This file must be modified to match your configuration.
+### Basic authentication
 
-#### Web Configuration
+This connector uses Basic authentication to communicate with OSIsoft and requires Basic authentication is enabled.
```
Comment (TomKimsey) [inline README.md:59]:
```suggestion
This connector uses basic authentication (RFC7617) to communicate with OSIsoft and requires basic authentication to be enabled.
```

File: README.md [hunk 14]
```diff
@@ -40,21 +54,18 @@ More information can be found here: [Ewon Historical Logging](https://developer.
 
 ## Getting Started
 
-### Configuration
-
-User configuration of the application parameters is done through the ConnectorConfig.json file.  This file must be modified to match your configuration.
+### Basic authentication
```
Comment (TomKimsey) [inline README.md:57]:
```suggestion
### Basic Authentication
```

File: README.md [hunk 15]
```diff
@@ -40,21 +54,18 @@ More information can be found here: [Ewon Historical Logging](https://developer.
...
 
-The ConnectorConfig.json file can be edited through the configuration webpage.  The configuration webpage can be accessed after the [installation process](#Installation) has been completed.  The url for the configuration interface **Your_Flexy's_IP_Address**/usr/config.html
+### Configuration
 
-![osisoft web configuration interface](docs/images/configurationPage.png)
+User configuration of the application parameters is done through the ConnectorConfig.json file.  This file must be modified to match your configuration.
 
 #### FTP Configuration
-If you wish to manually edit your ConnectorConfig.json file or you have a saved ConnectorConfig.json file from another Flexy you can load your ConnectorConfig.json file on the flexy in the /usr/ directory via FTP.
+To change configuration values, manually edit your ConnectorConfig.json file on the Flexy in the /usr/ directory via FTP.
```
Comment (TomKimsey) [inline README.md:66]:
```suggestion
To change configuration values, manually edit the ConnectorConfig.json file on the Flexy in the /usr/ directory via FTP.
```

Are you editing via FTP or updating the file via FTP and editing with another tool? FTP is not an editor.

File: README.md [hunk 16]
```diff
@@ -105,72 +116,109 @@ static String eWONCertificatePath = "/usr/Certificates";
...
+#### WebID
+This field is only needed when the CommunicationType is set to "piwebapi". WebID is the web ID of your OSIsoft PI Web API.
+This value can be attained by pointing a browser to the OSIsoft PIWEBAPI URL (https://192.168.0.124/piwebapi/dataservers).  The WebID is one of several parameters returned.  A username and password will be required to access the page.  If a username is not required it may be cached.  Try clearing the cache to verify the username and password.  The same username and password will be used to create the credentials in the next step. 
+(If the PIWEBAPI certificates are self-signed a security warning may occur when trying to access this page. This warning should be ignored)
```
Comment (TomKimsey) [inline README.md:148]:
```suggestion
(If the PIWEBAPI certificates are self-signed a security warning may occur when trying to access this page. This warning should be ignored.)
```

File: README.md [hunk 17]
```diff
@@ -105,72 +116,109 @@ static String eWONCertificatePath = "/usr/Certificates";
...
 
 #### Proxy URL
 
-Adding the field "ProxyURL" under the server config section will change request endpoints to https://domain-name/proxy-path. This field is optional. 
-Below is an example config file with the optional proxy URL in use.
-   ```
+This optional field is only needed when using a non default endpoint of https://HOSTNAME/piwebapi. Adding the field "ProxyURL" under the server config section will change request endpoints to https://HOSTNAME/proxy-path. HOSTNAME will be retrieved from the URL field in the configuration file.
```
Comment (TomKimsey) [inline README.md:151]:
```suggestion
This optional field is only needed when using a non-default endpoint of https://HOSTNAME/piwebapi. Adding the field "ProxyURL" under the server config section will change request endpoints to https://HOSTNAME/proxy-path. HOSTNAME will be retrieved from the URL field in the configuration file. 
```

File: README.md [hunk 18]
```diff
@@ -105,72 +116,109 @@ static String eWONCertificatePath = "/usr/Certificates";
...
+(If the PIWEBAPI certificates are self-signed a security warning may occur when trying to access this page. This warning should be ignored)
 
 #### Proxy URL
 
-Adding the field "ProxyURL" under the server config section will change request endpoints to https://domain-name/proxy-path. This field is optional. 
-Below is an example config file with the optional proxy URL in use.
-   ```
+This optional field is only needed when using a non default endpoint of https://HOSTNAME/piwebapi. Adding the field "ProxyURL" under the server config section will change request endpoints to https://HOSTNAME/proxy-path. HOSTNAME will be retrieved from the URL field in the configuration file. 
+
+Below is a partial config file with the optional proxy URL in use. Its endpoint is set to https://192.168.1.1/long/proxy/example/omf
```
Comment (TomKimsey) [inline README.md:153]:
```suggestion
Below is a partial config file with the optional proxy URL in use. Its endpoint is set to https://192.168.1.1/long/proxy/example/omf.
```

File: src/main/java/com/hms/flexyosisoftconnector/configuration/OSIsoftTagNamingScheme.java [hunk 19]
```diff
@@ -130,12 +130,17 @@ public static String extractTagNameFromComponents(String tagName, String tagType
...
+        if (index > 0) {
+          parsedTagName += optionNameDeliminator;
+        }
+        String shortHandOption = shortHandOSIsoftNameOptionSelected.substring(index, index + 2);
```
Comment (TomKimsey) [inline src/main/java/com/hms/flexyosisoftconnector/configuration/OSIsoftTagNamingScheme.java:143]: magic number

**Review by TomKimsey — CHANGES_REQUESTED** [review]

File: src/main/java/com/hms/flexyosisoftconnector/configuration/OSIsoftTagNamingScheme.java (hunk 19 above)
Comment (oliver-walker-hms) [reply src/main/java/com/hms/flexyosisoftconnector/configuration/OSIsoftTagNamingScheme.java]: Not using lengthOfSingleOption here is why there was an issue in the first place.. fixed

File: README.md [hunk 20]
```diff
@@ -132,23 +133,15 @@ This value can be attained by pointing a browser to the OSIsoft PIWEBAPI URL (ht
...
       "ProxyURL":"long/proxy/example/omf"
-   },
-   "eWONConfig":{
-      "CertificatePath":"/usr"
-   },
-   "AppConfig": {
-       "LoggingLevel": 4,
-       "CommunicationType": "omf"
-       }
+      },
```
Comment (alexjhawk) [inline README.md:162]: trailing comma should be removed. Also indentation is improper on the closing bracket

File: README.md (hunk 10 above)
Comment (alexjhawk) [reply README.md:159]: @oliver-walker-hms I would add a commit to fix indentation on JSON in README.md...looking at the JSON right now makes it a bit harder to follow with improper indentation on a few portions.

**Review by alexjhawk — CHANGES_REQUESTED** [review]

File: README.md (hunk 12 above)
Comment (TomKimsey) [reply README.md:63]: This got missed @oliver-walker-hms

**Review by alexjhawk — APPROVED** [review]

**Review by TomKimsey — APPROVED** [review]
